src/query/RelayFragmentPointer.js

```diff
@@ -66,11 +66,12 @@ class RelayFragmentPointer {
         });
       });
     }
+    let argType = typeof identifyingArgValue;
     invariant(
-      typeof identifyingArgValue === 'string' || identifyingArgValue == null,
+      typeof argType !== 'function' && argType !== 'symbol',
       'RelayFragmentPointer: Value for the argument to `%s` on query `%s` ' +
-      'should be a string, but it was set to `%s`. Check that the value is a ' +
-      'string.',
+      'should be boolean|string|number|object, but it was set to `%s`. ' +
+      'And this should not happen, must be some bugs in code of Relay',
```

Comment:
I think (hope) that it's more likely that someone will pass a function as an argument (perhaps doing `value: someFn` and forgetting to call the function) than that we'll write a bug here. Let's drop the last line ;-)